gcc/rust/backend/rust-mangle.h

```diff
@@ -0,0 +1,59 @@
+// This file is part of GCC.
+
+// GCC is free software; you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 3, or (at your option) any later
+// version.
+
+// GCC is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+// for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with GCC; see the file COPYING3.  If not see
+// <http://www.gnu.org/licenses/>.
+
+#ifndef RUST_MANGLE_H
+#define RUST_MANGLE_H
+
+#include "rust-compile-tyty.h"
+
+namespace Rust {
+namespace Compile {
+class Mangler
+{
+public:
+  enum MangleVersion
+  {
+    LEGACY = 0,
+    V0,
+  };
+
+  // this needs to support Legacy and V0 see github #429 or #305
+  std::string mangle_item (const TyTy::BaseType *ty,
+			   const Resolver::CanonicalPath &path,
+			   const std::string &crate_name) const;
+
+  std::string mangle_impl_item (const TyTy::BaseType *self,
+				const TyTy::BaseType *ty,
+				const std::string &name,
+				const std::string &crate_name) const;
+
+  static bool choose_mangling (std::string arg)
```

Comment:
Using Enum from https://gcc.gnu.org/onlinedocs/gccint/Option-properties.html would help here and save you from doing this string->enum variant by hand

Reply:
Done in 37f3b02

Reply:
I used an int for simplicity, with a static cast. I don't know how ugly that is in C++, but it was easier and simpler to read than declaring the enum somewhere other than in `rust-mangle.h`